src/socket.rs

```diff
@@ -956,6 +958,18 @@ impl Socket {
         self.inner().set_reuse_port(reuse)
     }
 
+    /// Set value for the `SO_BINDTODEVICE` option on this socket.
+    ///
+    /// If a socket is bound to an interface, only packets received from that
+    /// particular interface are processed by the socket. Note that this only
+    /// works for some socket types, particularly AF_INET sockets.
+    ///
+    /// This function is only available on Linux.
+    #[cfg(all(feature = "all", target_os = "linux"))]
+    pub fn set_bind_interface(&self, interface: CString) -> io::Result<()> {
```

Comment:
`CString` is an owned version, since we just need to use for a while, so we can use `CStr` (the borrowed version) instread.

Reply:
This should accept an `Option` where `None` which will set the length to zero, removing the interface binding. Or we could add a `remove_bind_interface` function which does this.